src/controllers/plugins/api/Message.cpp

```diff
@@ -163,7 +162,60 @@ std::unique_ptr<MessageElement> elementFromTable(const sol::table &tbl)
     assert(el);
 
     el->setTrailingSpace(tbl.get_or("trailing_space", true));
-    el->setTooltip(tbl.get_or("tooltip", QString{}));
+
+    auto link = tbl.get<sol::optional<Link>>("link");
+    if (link)
+    {
+        bool ok = false;
+        el->setLink(*link, &ok);
+        if (!ok)
+        {
+            throw std::runtime_error("'link' not supported on type='" +
+                                     type.toStdString() + '\'');
+        }
+        QString tooltip = tbl.get_or("tooltip", QString{});
+        if (!tooltip.isEmpty())
+        {
+            tooltip += "<br>";
+        }
+
+        switch (link->type)
+        {
+            case Link::Url:
```

Comment:
I'm not sure about this.

On the one hand I like this, because it shows the user what clicking would do, but on the other hand, this feels a bit too far - a plugin won't be able to really control the tooltip.

Also, with message modification, a plugin could overwrite this (mabye it could also inject some HTML to make the message disappear? Not sure).

An alternative could be to add some special marker to a link (that it came from a plugin) and have the channel-view show something special. Maybe even a special thing where holding shift would show the value of the link? Could be nice for regular links too.